### PR TITLE
Small improvements

### PR DESCRIPTION
**Pull request by slokhorst**
I added proper encoding of the login credentials, so if they contain symbols it will still work.
Also added the `logout` method as described in the spec. Might come in handy.

### CODE REVIEW

**Comment by ajalexander** [comment]
@lonaowna I'm getting errors when I test the 'logout' method you added. The message I'm getting is "Zlib::GzipFile::Error: not in gzip format".

However, it actually does 'logout' (such that when I make a subsequent call to 'status', I receive an authentication error).

File: lib/mmonit/connection.rb [hunk 1]
```diff
@@ -40,7 +40,11 @@ def connect
 		end
 
 		def login
-			self.request('/z_security_check', "z_username=#{@username}&z_password=#{@password}", true).code.to_i == 302
+			self.request('/z_security_check', URI.encode_www_form({'z_username'=>@username, 'z_password'=>@password, 'z_csrf_protection'=>'off'}), true).code.to_i == 302
+		end
+
+		def logout
```
Comment (ajalexander) [inline lib/mmonit/connection.rb:55]: I'm wondering whether this should also wipe out the @http variable.

Comment (slokhorst) [inline lib/mmonit/connection.rb:55]: I'm sorry, I didn't test the logout method properly. It seems to be that the 'logout' request returns an incorrectly compressed response. This is fixed by requesting an uncompressed response, by setting the 'Accept-Encoding: identity' header. (similar bug: https://github.com/jaimeiniesta/metainspector/issues/125)

Comment (slokhorst) [inline lib/mmonit/connection.rb:55]: And you're right, the logout method invalidates the zsessionid cookie retrieved in 'connect', so it should wipe @http to force a re-connect. Maybe we should call it 'disconnect' instead?

**Comment by ajalexander** [comment]
Do you plan to make that 'Accept-Encoding' change and test it out?

Rather than just renaming it, what do you think about having both a 'disconnect' and a 'logout' method? That would mirror the 'connect' and 'login' setup that currently exists. The 'disconnect' method would call 'logout' and would also clear the http variable.

**Comment by slokhorst** [comment]
I did commit and test the fixed logout method (f34f2c4337b3009bfce9063d5782b8f89e35faab). It works great now. The 'disconnect' method is a great idea. The logout method should not be called separately however, as it leaves the connection in an unusable state without an explicit reconnect. I have implemented and tested them both in e6d93cc04ee3a384cb08b19d2f05d2a153e7941e.

Thanks for all your comments!

**Comment by ajalexander** [comment]
I just retested this. It seems to work as expected now.

Thanks for the PR.

**Pull request merged**